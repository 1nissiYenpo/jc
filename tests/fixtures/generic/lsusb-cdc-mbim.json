[{"bus":"001","device":"003","id":"2c7c:0125","description":"Quectel Wireless Solutions Co., Ltd. EC25 LTE modem","device_descriptor":{"bLength":{"value":"18"},"bDescriptorType":{"value":"1"},"bcdUSB":{"value":"2.00"},"bDeviceClass":{"value":"239","description":"Miscellaneous Device"},"bDeviceSubClass":{"value":"2"},"bDeviceProtocol":{"value":"1","description":"Interface Association"},"bMaxPacketSize0":{"value":"64"},"idVendor":{"value":"0x2c7c","description":"Quectel Wireless Solutions Co., Ltd."},"idProduct":{"value":"0x0125","description":"EC25 LTE modem"},"bcdDevice":{"value":"3.18"},"iManufacturer":{"value":"1","description":"Quectel Incorporated"},"iProduct":{"value":"2","description":"LTE Module"},"iSerial":{"value":"0"},"bNumConfigurations":{"value":"1"},"configuration_descriptor":{"bLength":{"value":"9"},"bDescriptorType":{"value":"2"},"wTotalLength":{"value":"0x0109"},"bNumInterfaces":{"value":"6"},"bConfigurationValue":{"value":"1"},"iConfiguration":{"value":"0"},"bmAttributes":{"value":"0xa0","attributes":["(Bus Powered)","Remote Wakeup"]},"MaxPower":{"description":"500mA"},"interface_association":{"bLength":{"value":"8"},"bDescriptorType":{"value":"11"},"bFirstInterface":{"value":"4"},"bInterfaceCount":{"value":"2"},"bFunctionClass":{"value":"2","description":"Communications"},"bFunctionSubClass":{"value":"14"},"bFunctionProtocol":{"value":"0"},"iFunction":{"value":"0"}},"interface_descriptors":[{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"0"},"bAlternateSetting":{"value":"0"},"bNumEndpoints":{"value":"2"},"bInterfaceClass":{"value":"255","description":"Vendor Specific Class"},"bInterfaceSubClass":{"value":"255","description":"Vendor Specific Subclass"},"bInterfaceProtocol":{"value":"255","description":"Vendor Specific Protocol"},"iInterface":{"value":"0"},"endpoint_descriptors":[{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x81","description":"EP 1 IN"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x01","description":"EP 1 OUT"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}}]},{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"1"},"bAlternateSetting":{"value":"0"},"bNumEndpoints":{"value":"3"},"bInterfaceClass":{"value":"255","description":"Vendor Specific Class"},"bInterfaceSubClass":{"value":"0"},"bInterfaceProtocol":{"value":"0"},"iInterface":{"value":"0"},"** UNRECOGNIZED:  05":{"value":"24","description":"06 00 00"},"** UNRECOGNIZED:  04":{"value":"24","description":"02 02"},"endpoint_descriptors":[{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x83","description":"EP 3 IN"},"bmAttributes":{"value":"3","attributes":["Transfer Type  Interrupt","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x000a","description":"1x 10 bytes"},"bInterval":{"value":"9"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x82","description":"EP 2 IN"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x02","description":"EP 2 OUT"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}}]},{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"2"},"bAlternateSetting":{"value":"0"},"bNumEndpoints":{"value":"3"},"bInterfaceClass":{"value":"255","description":"Vendor Specific Class"},"bInterfaceSubClass":{"value":"0"},"bInterfaceProtocol":{"value":"0"},"iInterface":{"value":"0"},"** UNRECOGNIZED:  05":{"value":"24","description":"06 00 00"},"** UNRECOGNIZED:  04":{"value":"24","description":"02 02"},"endpoint_descriptors":[{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x85","description":"EP 5 IN"},"bmAttributes":{"value":"3","attributes":["Transfer Type  Interrupt","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x000a","description":"1x 10 bytes"},"bInterval":{"value":"9"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x84","description":"EP 4 IN"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x03","description":"EP 3 OUT"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}}]},{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"3"},"bAlternateSetting":{"value":"0"},"bNumEndpoints":{"value":"3"},"bInterfaceClass":{"value":"255","description":"Vendor Specific Class"},"bInterfaceSubClass":{"value":"0"},"bInterfaceProtocol":{"value":"0"},"iInterface":{"value":"0"},"** UNRECOGNIZED:  05":{"value":"24","description":"06 00 00"},"** UNRECOGNIZED:  04":{"value":"24","description":"02 02"},"endpoint_descriptors":[{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x87","description":"EP 7 IN"},"bmAttributes":{"value":"3","attributes":["Transfer Type  Interrupt","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x000a","description":"1x 10 bytes"},"bInterval":{"value":"9"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x86","description":"EP 6 IN"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x04","description":"EP 4 OUT"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}}]},{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"4"},"bAlternateSetting":{"value":"0"},"bNumEndpoints":{"value":"1"},"bInterfaceClass":{"value":"2","description":"Communications"},"bInterfaceSubClass":{"value":"14"},"bInterfaceProtocol":{"value":"0"},"iInterface":{"value":"6","description":"LTE Module"},"cdc_header":{"bcdCDC":{"value":"1.10"}},"cdc_union":{"bMasterInterface":{"value":"4"},"bSlaveInterface":{"value":"5"}},"cdc_mbim":{"bcdMBIMVersion":{"value":"1.00"},"wMaxControlMessage":{"value":"4096"},"bNumberFilters":{"value":"32"},"bMaxFilterSize":{"value":"128"},"wMaxSegmentSize":{"value":"2048"},"bmNetworkCapabilit":{"description":"0x20","attributes":["8-byte ntb input size"]}},"cdc_mbim_extended":{"bcdMBIMExtendedVersion":{"value":"1.00"},"bMaxOutstandingCommandMessages":{"value":"64"},"wMTU":{"value":"1500"}},"endpoint_descriptors":[{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x89","description":"EP 9 IN"},"bmAttributes":{"value":"3","attributes":["Transfer Type  Interrupt","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0040","description":"1x 64 bytes"},"bInterval":{"value":"9"}}]},{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"5"},"bAlternateSetting":{"value":"0"},"bNumEndpoints":{"value":"0"},"bInterfaceClass":{"value":"10","description":"CDC Data"},"bInterfaceSubClass":{"value":"0"},"bInterfaceProtocol":{"value":"2"},"iInterface":{"value":"7","description":"MBIM Data"}},{"bLength":{"value":"9"},"bDescriptorType":{"value":"4"},"bInterfaceNumber":{"value":"5"},"bAlternateSetting":{"value":"1"},"bNumEndpoints":{"value":"2"},"bInterfaceClass":{"value":"10","description":"CDC Data"},"bInterfaceSubClass":{"value":"0"},"bInterfaceProtocol":{"value":"2"},"iInterface":{"value":"7","description":"MBIM Data"},"endpoint_descriptors":[{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x88","description":"EP 8 IN"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}},{"bLength":{"value":"7"},"bDescriptorType":{"value":"5"},"bEndpointAddress":{"value":"0x05","description":"EP 5 OUT"},"bmAttributes":{"value":"2","attributes":["Transfer Type  Bulk","Synch Type  None","Usage Type  Data"]},"wMaxPacketSize":{"value":"0x0200","description":"1x 512 bytes"},"bInterval":{"value":"0"}}]}]}},"device_qualifier":{"bLength":{"value":"10"},"bDescriptorType":{"value":"6"},"bcdUSB":{"value":"2.00"},"bDeviceClass":{"value":"239","description":"Miscellaneous Device"},"bDeviceSubClass":{"value":"2"},"bDeviceProtocol":{"value":"1","description":"Interface Association"},"bMaxPacketSize0":{"value":"64"},"bNumConfigurations":{"value":"1"}},"device_status":{"value":"0x0000","description":"(Bus Powered)"}}]
