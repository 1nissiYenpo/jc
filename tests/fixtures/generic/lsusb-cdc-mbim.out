Bus 001 Device 003: ID 2c7c:0125 Quectel Wireless Solutions Co., Ltd. EC25 LTE modem
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass          239 Miscellaneous Device
  bDeviceSubClass         2 
  bDeviceProtocol         1 Interface Association
  bMaxPacketSize0        64
  idVendor           0x2c7c Quectel Wireless Solutions Co., Ltd.
  idProduct          0x0125 EC25 LTE modem
  bcdDevice            3.18
  iManufacturer           1 Quectel Incorporated
  iProduct                2 LTE Module
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0109
    bNumInterfaces          6
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass    255 Vendor Specific Subclass
      bInterfaceProtocol    255 Vendor Specific Protocol
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval               9
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x85  EP 5 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval               9
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x03  EP 3 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        3
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x87  EP 7 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval               9
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x86  EP 6 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x04  EP 4 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
    Interface Association:
      bLength                 8
      bDescriptorType        11
      bFirstInterface         4
      bInterfaceCount         2
      bFunctionClass          2 Communications
      bFunctionSubClass      14 
      bFunctionProtocol       0 
      iFunction               0 
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        4
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         2 Communications
      bInterfaceSubClass     14 
      bInterfaceProtocol      0 
      iInterface              6 LTE Module
      CDC Header:
        bcdCDC               1.10
      CDC Union:
        bMasterInterface        4
        bSlaveInterface         5 
      CDC MBIM:
        bcdMBIMVersion       1.00
        wMaxControlMessage   4096
        bNumberFilters       32
        bMaxFilterSize       128
        wMaxSegmentSize      2048
        bmNetworkCapabilities 0x20
          8-byte ntb input size
      CDC MBIM Extended:
        bcdMBIMExtendedVersion           1.00
        bMaxOutstandingCommandMessages     64
        wMTU                             1500
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x89  EP 9 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               9
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        5
      bAlternateSetting       0
      bNumEndpoints           0
      bInterfaceClass        10 CDC Data
      bInterfaceSubClass      0 
      bInterfaceProtocol      2 
      iInterface              7 MBIM Data
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        5
      bAlternateSetting       1
      bNumEndpoints           2
      bInterfaceClass        10 CDC Data
      bInterfaceSubClass      0 
      bInterfaceProtocol      2 
      iInterface              7 MBIM Data
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x88  EP 8 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x05  EP 5 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
Device Qualifier (for other device speed):
  bLength                10
  bDescriptorType         6
  bcdUSB               2.00
  bDeviceClass          239 Miscellaneous Device
  bDeviceSubClass         2 
  bDeviceProtocol         1 Interface Association
  bMaxPacketSize0        64
  bNumConfigurations      1
Device Status:     0x0000
  (Bus Powered)
